### DISCUSSION TITLE
Why does microsoft ignore all the winui performance complaints?

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/microsoft-ui-xaml#8625] [General]
### Describe the bug

This is very unprofessional behavior when you make a UI framework and just straight up ignore all the complaints about performance which is something very critical when you base windows 11 apps off of it!

https://github.com/microsoft/microsoft-ui-xaml/issues/4713
https://github.com/microsoft/microsoft-ui-xaml/issues/2032

this benchmark backs up my claims as well
https://github.com/Noemata/XamlBenchmark

even users are experiencing complaints with proof:
https://www.reddit.com/r/windowsinsiders/comments/116yxtn/whys_task_manager_so_hecin_slow/

### Steps to reproduce the bug

Use any windows 11 app

### NuGet package version

None

**Comment by abart27** [comment]
> even users are experiencing complaints with proof:
https://www.reddit.com/r/windowsinsiders/comments/116yxtn/whys_task_manager_so_hecin_slow/

The Windows 11 Task Manager is UWP + WinUI2 hosted in a Win32 application via a XAML Island.
Those performance issues aren't related to WinUI3.

**Comment by ghost** [comment]
> > even users are experiencing complaints with proof:
> > https://www.reddit.com/r/windowsinsiders/comments/116yxtn/whys_task_manager_so_hecin_slow/
> 
> The Windows 11 Task Manager is UWP + WinUI2 hosted in a Win32 application via a XAML Island. Those performance issues aren't related to WinUI3.

so the performance issues are for winui 2?

**Comment by abart27** [comment]
No, WinUI2 is a visual styling library and doesn't have any inherent performance issues.

I can't comment on the internals and causes of the poor performance, but it's safe to assume that UWP is not the bottleneck here.

**Comment by DarranRowe** [comment]
> No, WinUI2 is a visual styling library and doesn't have any inherent performance issues.

This isn't true. WinUI 2 is also a controls library. Compare the WinUI 2 [Microsoft.UI.Xaml.Controls](https://learn.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls?view=winui-2.8) namespace with [Windows.UI.Xaml.Controls](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls?view=winrt-22621).

**Comment by tusharsnx** [comment]
TaskManager isn't slow! 

The linked reddit post seems to mention that the TaskManager is slow to expand sidebar. 

Reason:
The legacy view isn't animate 'able, so they artificially applied a delay before shifting the UI to account for the expanded sidebar. When you expand the sidebar, it animates to its full width. You can check any other WinUi app that has a sidebar to get an idea. The sidebar (animates and) expands behind the legacy view, and once it fully expanded, it is put in front at once after the delay. The legacy UI shrinks to account for the expanded sidebar. How I know this? Well, there was a time when I could see the slightest part of the sidebar growing behind the base. I had filed feedback then. I think they've fixed it now.

So, it isn't slow, but rather an implementation 'defect'. But, I'm also not sure if there's any better solution in this case. A rewrite of whole UI might be the solution, but who knows how much time that would take.
